builder/tests/test_views.py

```diff
@@ -379,3 +380,92 @@ def test_search_length_validation(client, draft_with_no_searches):
+    min_search_term_length = get_validator_limit_value(
+        term_field_validators, MinLengthValidator
+    )
+
+    term_with_min_chars = "a" * min_search_term_length
+    term_with_too_few_chars = "a" * (min_search_term_length - 1)
+    expected_term_error = (
+        f"Ensure this value has at least {min_search_term_length} characters".encode()
+    )
+
+    # If the search term length is at the limit then it is a valid search so the
+    # search count increments by one and the error message is not in the response
+    with assert_difference(
+        draft_with_no_searches.searches.count, expected_difference=1
+    ):
+        rsp = client.post(
+            draft_with_no_searches.get_builder_new_search_url(),
+            {"search": term_with_min_chars, "search-type": "term"},
+            follow=True,
+        )
+    assert rsp.status_code == 200
+    assert expected_term_error not in rsp.content
+
+    # If the search term length is under the limit then it is an invalid search so
+    # the search count does not increment and the error message is in the response
+    with assert_difference(
+        draft_with_no_searches.searches.count, expected_difference=0
+    ):
+        rsp = client.post(
+            draft_with_no_searches.get_builder_new_search_url(),
+            {"search": term_with_too_few_chars, "search-type": "term"},
+            follow=True,
+        )
+    assert rsp.status_code == 200
+    assert expected_term_error in rsp.content
+
+    code_field_validators = Search._meta.get_field("code").validators
+    min_search_code_length = get_validator_limit_value(
+        code_field_validators, MinLengthValidator
+    )
+    code_with_min_chars = "a" * min_search_code_length
+    expected_code_error = (
+        f"Ensure this value has at least {min_search_code_length} characters".encode()
+    )
+
+    # If the search code length is at the limit then it is a valid search so the
+    # search count increments by one and the error message is not in the response
+    with assert_difference(
+        draft_with_no_searches.searches.count, expected_difference=1
+    ):
+        rsp = client.post(
+            draft_with_no_searches.get_builder_new_search_url(),
+            {"search": code_with_min_chars, "search-type": "code"},
+            follow=True,
+        )
+    assert rsp.status_code == 200
+    assert expected_code_error not in rsp.content
+
+    # Note: The current minimum search code length is 1.
+    # Below is an assertion to validate this.
+    # A minimum search code length of 1 means the value
+    # below the minimum is zero.
+    # What happens if we enter a search below the minimum
+    # code length is that we encounter the case where no
+    # no search is entered. That case is handled differently
+    # by new_search() to guard against searches
+    # that cannot be slugified.
+    # We do not reach the Django MinValidationError,
+    # unlike for terms.
+    assert len(code_with_min_chars) == 1
```

Comment:
thought: This part being slightly unusual does feel a little bit like a hint the code is more complex than it could be. But I think this is OK, because the function of the slugify check is a special case to ensure there are valid characters.